Framework/DataHandling/test/LoadInstrumentTest.h

```diff
@@ -389,12 +378,13 @@ class LoadInstrumentTest : public CxxTest::TestSuite {
     TS_ASSERT(!physicalInst->isParametrized());
 
     // Check the positions of the 6 detectors in the physical instrument
-    TS_ASSERT_EQUALS(physicalInst->getDetector(1000)->getPos(), V3D(0, 0, 0));
-    TS_ASSERT_EQUALS(physicalInst->getDetector(1001)->getPos(), V3D(0, 1, 0));
-    TS_ASSERT_EQUALS(physicalInst->getDetector(1002)->getPos(), V3D(1, 0, 0));
-    TS_ASSERT_EQUALS(physicalInst->getDetector(1003)->getPos(), V3D(1, 1, 0));
-    TS_ASSERT_EQUALS(physicalInst->getDetector(1004)->getPos(), V3D(2, 0, 0));
-    TS_ASSERT_EQUALS(physicalInst->getDetector(1005)->getPos(), V3D(2, 1, 0));
+    const auto &physicalDetectorInfo = ws->detectorInfo();
+    TS_ASSERT_EQUALS(physicalDetectorInfo.position(1000), V3D(0, 0, 0));
```

Comment:
This `physicalDetectorInfo.position(1000)` and the ones below also make the test crash.